maskrcnn_benchmark/structures/segmentation_mask.py

```diff
@@ -8,63 +11,207 @@
 FLIP_TOP_BOTTOM = 1
 
 
-class Mask(object):
+""" ABSTRACT
+Segmentations come in either:
+1) Binary masks
+2) Polygons
+
+Binary masks can be represented in a contiguous array
+and operations can be carried out more efficiently,
+therefore BinaryMaskList handles them together.
+
+Polygons are handled separately for each instance,
+by PolygonInstance and instances are handled by
+PolygonList.
+
+SegmentationList is supposed to represent both,
+therefore it wraps the functions of BinaryMaskList
+and PolygonList to make it transparent.
+"""
+
+
+class BinaryMaskList(object):
     """
-    This class is unfinished and not meant for use yet
-    It is supposed to contain the mask for an object as
-    a 2d tensor
+    This class handles binary masks for all objects in the image
     """
 
-    def __init__(self, masks, size, mode):
+    def __init__(self, masks, size):
+        """
+            Arguments:
+                masks: Either torch.tensor of [num_instances, H, W]
+                    or list of torch.tensors of [H, W] with num_instances elems,
+                    or RLE (Run Length Encoding) - interpreted as list of dicts,
+                    or BinaryMaskList.
+                size: absolute image size, width first
+
+            After initialization, a hard copy will be made, to leave the
+            initializing source data intact.
+        """
+
+        if isinstance(masks, torch.Tensor):
+            # The raw data representation is passed as argument
+            masks = masks.clone()
+        elif isinstance(masks, (list, tuple)):
+            if isinstance(masks[0], torch.Tensor):
+                masks = torch.stack(masks, dim=2).clone()
+            elif isinstance(masks[0], dict) and "count" in masks[0]:
+                # RLE interpretation
+
+                masks = mask_utils
```

Comment:
Shouldn't this be different for the case of a list of RLEs?

Reply:
cc @botcs

Reply:
@IssamLaradji Nice catch!

Here is a quick guide for usage (and for sanity check on update):
https://gist.github.com/botcs/a59f3f59e22e5df93e3e5e4f86718af3

Reply:
Fixed in PR #657

Reply:
Perfect! the updated version looks good!

Reply:
Quick question, will this get merged into master :P?

Reply:
:P cc @fmassa can you merge #657?

Thanks